packages/babel-helper-create-class-features-plugin/src/fields.ts

```diff
@@ -449,16 +449,31 @@ function buildPrivateFieldInitLoose(ref, prop, privateNamesMap) {
   `;
 }
 
-function buildPrivateInstanceFieldInitSpec(ref, prop, privateNamesMap) {
+function buildPrivateInstanceFieldInitSpec(ref, prop, privateNamesMap, state) {
   const { id } = privateNamesMap.get(prop.node.key.id.name);
   const value = prop.node.value || prop.scope.buildUndefinedNode();
 
-  return template.statement.ast`${t.cloneNode(id)}.set(${ref}, {
-    // configurable is always false for private elements
-    // enumerable is always false for private elements
-    writable: true,
-    value: ${value},
-  })`;
+  const helper = state.addHelper("classPrivateFieldInitSpec");
+  const expr = t.callExpression(helper, [
+    t.thisExpression(),
+    t.cloneNode(id),
+    template.expression.ast`{
+        writable: true,
+        value: ${value}
+      }`,
+  ]);
```

Comment:
This plugin could be used with older `@babel/core` versions, that don't support the new `classPrivateFieldInitSpec` helper. We should do something like this:
```js
if (state.availableHelper("classPrivateFieldInitSpec")) {
  // use the helper
} else {
  // return the old output, even if partially incorrect
}
```

Reply:
Done!